semtypes/src/main/java/io/ballerina/types/definition/ListDefinition.java

```diff
@@ -55,37 +65,40 @@ public SemType getSemType(Env env) {
         }
     }
 
+    public static SemType tupleTypeWrapped(Env env, SemType... members) {
+        ListDefinition def = new ListDefinition();
+        return def.define(env, List.of(members), members.length);
+    }
+
+    public static SemType tupleTypeWrappedRo(Env env, SemType... members) {
+        ListDefinition def = new ListDefinition();
+        return def.define(env, List.of(members), members.length, NEVER, CELL_MUT_NONE);
+    }
+
     // Overload define method for commonly used default parameter values
 
-    /***
-     * Define a tuple type without a rest type.
-     */
-    public ComplexSemType define(Env env, List<SemType> initial) {
-        return define(env, initial, initial.size(), PredefinedType.NEVER);
+    public SemType define(Env env, List<SemType> initial, int size) {
```

Comment:
these are overloaded methods of `defineListTypeWrapped()` API right? Shall we fix the naming?

Reply:
Methods that were called `define` are instance methods that modify the ListDefinition where as `defineListTypeWrapped` (and it tuple counterparts) are stantic methods that create their own "local" list definitions. Since calling both `define` maybe confusing I decided to rename former to resolve, WDYT?

Reply:
Let's update according to our offline discussion. :)

Reply:
Fixed with one slight modification "main" `defineListTypeWrapped` converted to instance method since in most places we use overrides (like BArrayType) we need an instance method (or need to directly call `define` which   IMO is bad since it expose the fact we use cells to outside) and having both a static and instance methods with same name was deemed confusing